File: game.ml

type game = {
  players: Player.player list;
    board: Board.board;
}

let game_over (game:game) : bool =
  List.exists (fun p -> Player.score_for_player p >= 15) game.players

module type Game = sig
  val game_over : game -> bool
end

module DefaultGame : Game = struct
  let game_over (game:game) : bool =
    List.exists (fun p -> Player.score_for_player p >= 15) game.players
end

module QuickGame : Game = struct
  let game_over (game:game) : bool =
    List.exists (fun p -> Player.score_for_player p >= 5) game.players
end

module DebugGame : Game = struct
  let game_over (game:game) : bool =
    List.exists (fun p -> List.length p.Player.cards > 0) game.players
end

File: python/game.py

from board import Board
from player import Player

class Game(object):
    def __init__(self, players, board):
        self.players = players
        self.board = board

    @staticmethod
    def default_game():
        player_count = int(raw_input('Number of players: '))
        players = []
        for i in xrange(player_count):
            players.append(Player(raw_input('  Name of Player #%d: ' % (i+1,))))
        return Game(players, Board.default_board())
